pages/custom-query-modules/rust/rust-example.mdx

```diff
@@ -0,0 +1,214 @@
+---
+title: Example of a query module written in Rust
+description: Discover how to develop and implement a Rust query module in Memgraph.
+---
+
+import { Callout } from 'nextra/components'
+import { Steps } from 'nextra/components'
+
+# Quickstart to custom query module in Rust
+
+In order to start working on a query module, you need to have a Memgraph instance running on a server or in a container. The container or the server where the instance is running will also be used as a development environment for your query module in this case. 
+
+This will provide you with an isolated environment for testing and developing your query module. In order to follow this guide, the best way would be to have Docker installed on your machine.
+
+- [Docker](https://docs.docker.com/get-docker/)
+
+
+If you do not use Docker but have Memgraph on your Linux machine, the following steps will be the same. You will just run commands on your machine instead of the docker container. However,
+you'll need to ensure that system packages required to compile the query modules are present on your machine.
+
+<Steps>
+### Run Memgraph
+
+First, you need to run the Memgraph instance. The easiest way to create and build new query modules is inside our 
+`memgraph-mage-dev` Docker image, since it contains necessary system packages. 
+You can run the container with the following command:
+
+```bash
+docker run -p 7687:7687 -p 7444:7444 --name mage memgraph/memgraph-mage:<version>-dev
+```
+
+### Enter into memgraph container shell
+
+After you have started the Memgraph instance, you need to enter the shell. You can do this by running the following command:
+
+```bash
+docker exec -it -u root mage bash
+```
+
+This will start bash inside the container as the root user. Since we will need to install certain dependencies, we need to be the root user. Of course, if you are running a native Linux machine, you can skip this step.
+
+### Install Cargo for Rust
+
+Query modules in Rust are compiled inside the Memgraph image using `cargo`. Cargo has a large overhead on the Docker image and it is not
+installed by default. In order to install cargo and add it to the `PATH` environment variable, we will use the following commands:
```

Comment:
How much time will this take? If more than a few seconds, please let the user know they will need to wait.